<launch>
  <arg name="usb_port"                default="/dev/ttyUSB0"/>
  <arg name="dxl_baud_rate"           default="1000000"/>
  <arg name="namespace"               default="red_manipulator"/>

  <arg name="use_moveit"              default="true"/>
  <arg name="use_joint_state"         default="true"/>
  <arg name="use_cmd_vel"             default="false"/>

  <param name="dynamixel_info"          value="$(find dynamixel_workbench_controllers)/config/arm.yaml"/>

  <node name="$(arg namespace)" pkg="dynamixel_workbench_controllers" type="dynamixel_workbench_controllers"
        required="true" output="screen" args="$(arg usb_port) $(arg dxl_baud_rate)">
    <param name="use_moveit"              value="$(arg use_moveit)"/>
    <param name="use_joint_states_topic"  value="$(arg use_joint_state)"/>
    <param name="use_cmd_vel_topic"       value="$(arg use_cmd_vel)"/>
    <!--remap from="/red_manipulator/joint_trajectory" to="/red_manipulator/arm_controller/command"/-->
    <rosparam>
      publish_period: 0.010
      dxl_read_period: 0.010
      dxl_write_period: 0.010
    </rosparam>
  </node>

  <rosparam file="$(find dynamixel_workbench_controllers)/config/arm_controller.yaml" command="load"/>

  <node ns="red_manipulator" name="trajectory_republisher" pkg="dynamixel_workbench_controllers" type="trajectory_republisher.py"
        required="true" output="screen"/>

</launch>
